runtime/include/bpftime_prog.hpp

```diff
@@ -13,7 +13,11 @@
 namespace bpftime
 {
 
+#ifdef __APPLE__
+extern __thread std::optional<uint64_t> current_thread_bpf_cookie;
```

Comment:
Why using __thread instead of thread_local?

Reply:
thread_local was not present in APPLE, i was giving errors when trying to compile it.

Reply:
> thread_local was not present in APPLE, i was giving errors when trying to compile it.

`thread_local` is a keyword introduced in C++11, you may need to check your compiler

https://en.cppreference.com/w/cpp/keyword/thread_local